query-tee: Add support for sending only a proportion of requests to secondary backends (#8532)
grafana/mimir: docs/sources/mimir/manage/tools/query-tee.md

- If the backend endpoint URL is configured only with a username, then query-tee keeps the configured username and injects the password received in the incoming request.
- If the backend endpoint URL is configured without a username and password, then query-tee forwards the authentication credentials found in the incoming request.

### Backend selection

> The query-tee supports always sending requests to all backends, or sending only a proportion of requests to all backends, with the remainder sent only to the preferred backend.
> This is configured with `-proxy.secondary-backends-request-proportion`.

Reviewer: ```suggestion
You can use the query-tee to either send requests to all backends or to send a proportion of requests to all backends and the remaining requests to your preferred backend.
Configure backend selection in the `-proxy.secondary-backends-request-proportion` setting.
```

Reviewer: Thank you for making these updates! I left a really minor comment, but these changes look great to me!

> For example, if `-proxy.secondary-backends-request-proportion` is 1.0, then all requests are sent to all backends.
> Or if `-proxy.secondary-backends-request-proportion` is 0.2, then 20% of requests are sent to all backends, and the remaining 80% are sent to the preferred backend only.

Reviewer: ```suggestion
For example, if you set `-proxy.secondary-backends-request-proportion` to `1.0`, then all requests are sent to all backends.
Alternatively, if you set `-proxy.secondary-backends-request-proportion` to `0.2`, then 20% of requests are sent to all backends and the remaining 80% of requests are only sent to your preferred backend.
```

Author: Resolved in 82af84c

### Backend selection

You can use the query-tee to either send requests to all backends, or to send a proportion of requests to all backends and the remaining requests to only the preferred backend.

> You can configure this with `-proxy.secondary-backends-request-proportion`.

Reviewer: I like specifying what `-proxy.secondary-backends-request-proportion` is a setting. For example, "You can configure this with the `-proxy.secondary-backends-request-proportion` setting. Otherwise, looks greats!

Author: Makes sense, I've updated it to be clearer. I've used the term "CLI flag" rather than "setting" as that's more specific and appropriate for this case.

For example, if you set `-proxy.secondary-backends-request-proportion` to `1.0`, then all requests are sent to all backends.

> Alternatively, if you set `-proxy.secondary-backends-request-proportion` to `0.2`, then 20% of requests are sent to all backends, and the remaining 80% of requests are sent only to your preferred backend.

Reviewer: Same comment as above. I think it's worth specifying that `-proxy.secondary-backends-request-proportion` is a setting.
